test/functional/web/safari-nativewebtap-e2e-specs.js

```diff
@@ -1,270 +1,268 @@
+      // default to a relatively representative set of devices
+      devices = [DEVICE_NAME, 'iPad Simulator'];
+    }
+
+    async function loadPage (driver, url) {
+      await retryInterval(5, 1000, async function () {
+        await openPage(driver, url);
+        await spinTitle(driver).should.eventually.not.include('Cannot Open Page');
+      });
+    }
+
+    // define the tests, for each device
+    for (const deviceName of devices) {
+      describe(`${deviceName} -`, function () {
+        this.timeout(MOCHA_TIMEOUT * 2);
+
+        let driver;
+        const localCaps = amendCapabilities(caps, {'appium:deviceName': deviceName});
+        let skipped = false;
+
+        before(async function () {
+          skipped = false;
+          try {
+            driver = await initSession(localCaps);
+          } catch (err) {
+            if (err.message.includes('Invalid device type') || err.message.includes('Incompatible device')) {
+              skipped = true;
+              return this.skip();
+            }
+            throw err;
+          }
+          if (process.env.CI) {
+            await driver.setTimeouts(10000);
+          }
+        });
+        after(async function () {
+          await deleteSession();
+        });
+        beforeEach(async function () {
+          await driver.updateSettings({
+            nativeWebTapStrict: false,
+          });
+        });
+
+        it('should be able to tap on an element', async function () {
+          await loadPage(driver, GUINEA_PIG_PAGE);
+
+          await driver.$(`=${PAGE_3_LINK}`).click();
+
+          await spinTitleEquals(driver, PAGE_3_TITLE, SPIN_RETRIES);
+        });
+
+        it('should be able to tap on an element when the app banner is up', async function () {
+          await loadPage(driver, GUINEA_PIG_APP_BANNER_PAGE);
+
+          await driver.$(`=${PAGE_3_LINK}`).click();
+
+          await spinTitleEquals(driver, PAGE_3_TITLE, SPIN_RETRIES);
+        });
+
+        it('should be able to bypass measuring the offset of banner', async function () {
+          await driver.updateSettings({
+            nativeWebTapStrict: true,
+          });
+
+          const start1 = performance.now();
+          await loadPage(driver, GUINEA_PIG_APP_BANNER_PAGE);
+          await driver.$(`=${PAGE_3_LINK}`).click();
+          await spinTitleEquals(driver, PAGE_3_TITLE, SPIN_RETRIES);
+          const end1 = performance.now();
+          const durationWithoutIgnore = end1 - start1;
+
+          const start2 = performance.now();
+          await driver.updateSettings({ nativeWebTapSmartAppBannerVisibility: 'invisible' });
+          await loadPage(driver, GUINEA_PIG_APP_BANNER_PAGE);
+          await driver.$(`=${PAGE_3_LINK}`).click();
+          await spinTitleEquals(driver, PAGE_3_TITLE, SPIN_RETRIES);
+          const end2 = performance.now();
+          const durationWithIgnore = end2 - start2;
+          durationWithIgnore.should.be.below(durationWithoutIgnore);
```

Comment:
I'm not sure if we should keep this assertion because I sometimes saw an error as
```
  1) iPad simulator -
       should be able to bypass measuring the offset of banner:

      AssertionError: expected 2773.466542005539 to be below 2744.7962079644203
      + expected - actual

      -2773.466542005539
      +2744.7962079644203
      
      at Context.<anonymous> (test/functional/web/safari-nativewebtap-e2e-specs.js:135:40)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

```

Reply:
yep, this might be unstable due to CI slowness
you could add `if process.env.CI` check

Reply:
thank you for your suggestion. I'd like to simply remove performance check part as https://github.com/appium/appium-xcuitest-driver/pull/1490/commits/aecffc09132ced4b6d4753d07c3feb7a95629ee7